dapr/actor/runtime/state_manager.py

```diff
@@ -69,6 +70,9 @@ def __init__(self, actor: 'Actor'):
         self._type_name = actor.runtime_ctx.actor_type_info.type_name
 
         self._default_state_change_tracker: Dict[str, StateMetadata] = {}
+        self._mock_state: Dict[str, Any]
```

Comment:
Do we need this?

Reply:
If you don't add this, then because of the type hinting problems mentioned earlier, the _mock_state, _mock_reminders and _mock_timer variables won't be correctly type hinted by the IDE, and will show up as white unknown variables.

With those type hint stubs
![image](https://github.com/user-attachments/assets/033bcd8d-bd9d-43f5-b6da-4c9f5cd41cbf)

Without

![image](https://github.com/user-attachments/assets/52380dc5-3453-4957-8388-a1195baef814)

Yes the code would still run just fine, but it's not very nice, no?

Type hinting the MockStateManager is of no use, because (as you can see in the image) it thinks the _state_manager variable it is dealing with is an ActorStateManager instead of MockStateManager.

In my view adding these stubs to ActorStateManager is harmless (at least, I cannot think of a scenario where they might cause problems or be undesirable) so I think they're worth having.

Reply:
I'm leaning more towards removing the `_mock` arguments from the main code, and rely on good documentation (which @lor1113  already did) for awareness of the properties' existence. What do you think @berndverst ?